src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js

```diff
@@ -62,16 +64,44 @@ const ActivityThread = ({
     repliesTotalCount = 0,
     translations,
 }: Props) => {
-    const [isExpanded, setIsExpanded] = React.useState(isAlwaysExpanded);
+    const { length: repliesLength } = replies;
+    const repliesToLoadCount = Math.max(repliesTotalCount - repliesLength, 0);
 
-    const toggleButtonLabel = isExpanded ? messages.hideReplies : messages.showReplies;
-    const repliesToLoadCount = Math.max(repliesTotalCount - 1, 0);
+    const onHideRepliesHandler = () => {
+        onHideReplies(replies[repliesLength - 1]);
+    };
+
+    const renderButton = () => {
+        if (isAlwaysExpanded || isRepliesLoading) {
+            return null;
+        }
 
-    const toggleReplies = () => {
-        if (!isExpanded) {
-            onShowReplies();
+        if (repliesTotalCount > repliesLength) {
+            return (
+                <PlainButton
+                    className="bcs-ActivityThread-toggle"
+                    onClick={onShowReplies}
+                    type="button"
+                    data-testid="activity-thread-button"
+                >
+                    <FormattedMessage values={{ repliesToLoadCount }} {...messages.showReplies} />
+                </PlainButton>
+            );
+        }
+        if (repliesTotalCount > 1 && repliesTotalCount === repliesLength) {
```

Comment:
So if replies total count is 1, we wont show any button? We should add a test for this too

Reply:
Correct. There was a test covering that, but I've made it now clearer to see.